connection.go

```diff
@@ -506,6 +515,8 @@ func (c *Connection) writeLoop() {
 	var err error
 
 	for err == nil {
+		idleTimeTimer := time.NewTimer(c.Opts.IdleTime)
```

Comment:
I think we can fo something like this


```suggestion
	idleTimeTimer := time.NewTimer(c.Opts.IdleTime)
	defer idleTimeTimer.Stop()
	for err == nil {
```

and then, inside the `for loop` reset the timer: `idleTimeTimer.Reset(c.Opts.IdleTime)`

Reply:
I think in some cases, it is possible than `case req := <-c.requestsCh:` & `case <-idleTimeTimer.C:` are ready in almost same time (or more specifically that `case <-idleTimeTimer.C:` is ready before `idleTimeTimer.Reset(c.Opts.IdleTime)`).
So in this case, in the next loop, we will take immediately `case <-idleTimeTimer.C` (because the chan `idleTimeTimer.C` is already ready to be read).
This is reason why I prefer to discard the timer and create a new one.

Reply:
Maybe you can explicitly drain the chan as discribed into godoc:

 ```
	if !t.Stop() {
		<-t.C
	}
	t.Reset(d)
```

Reply:
II need to think about it :)

Reply:
Maybe, we also encapsulate drain with select. Because`t.C` can be already drained (if expired), w/o select it will be deadlock here:

```go
   if !t.Stop() {
   	select{
   	case <-t.C:
   	default:
   	}
   }
   t.Reset(d)
```

Reply:
Maybe, a drainTimer helperfunc can be defined:

```go
func drainTimer(t *time.Timer) {
	if !t.Stop() {
		select {
		case <-t.C:
		default:
		}
	}
}
```

Reply:
or `resetTimer` helperfunc ?

```go
func ResetTimer(t *time.Timer, d time.Duration) {
	if !t.Stop() {
		select {
		case <-t.C:
		default:
		}
	}

	t.Reset(d)
}
```